snf-image-helper/inject-files.py

```diff
@@ -39,44 +40,58 @@ def timestamp():
     return current_time
 
 
-def parse_arguments(input_args):
-    usage = "Usage: %prog [options] <target>"
-    parser = OptionParser(usage=usage)
-    parser.add_option("-i", "--input",
-                      action="store", type='string', dest="input_file",
-                      help="get input from FILE instead of stdin",
-                      metavar="FILE")
-    parser.add_option("-d", "--decode", action="store_true", dest="decode",
-                      default=False,
-                      help="decode files under target and create manifest")
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-t", "--target", dest="target", metavar="DIRECTORY",
+                        help="write files to DIRECTORY")
+    parser.add_argument("-i", "--input", dest="input_file", metavar="FILE",
+                        help="get input from FILE instead of stdin")
+    parser.add_argument("-d", "--decode", action="store_true", dest="decode",
+                        default=False,
+                        help="decode files under target and create manifest")
+    parser.add_argument("-c", "--cloud-init", default=False, dest="cloudinit",
+                        action="store_true",
+                        help="decode files to a cloud-init configureation")
 
-    opts, args = parser.parse_args(input_args)
+    args = parser.parse_args()
 
-    if len(args) != 1:
-        parser.error('target is missing')
+    if args.target is not None and not os.path.isdir(args.target):
+        parser.error('Defined target: %s is not a directory' % args.target)
 
-    target = args[0]
-    if not os.path.isdir(target):
-        parser.error('target is not a directory')
+    if args.input_file is not None and not os.path.isfile(args.input_file):
+        parser.error('input file does not exist')
 
-    input_file = opts.input_file
-    if input_file is None:
-        input_file = sys.stdin
-    else:
-        if not os.path.isfile(input_file):
-            parser.error('input file does not exist')
-        input_file = open(input_file, 'r')
+    if args.target and args.cloudinit:
+        parser.error('Options -t and -c are mutual exclusive')
 
-    return (input_file, target, opts.decode)
+    if not args.target and not args.cloudinit:
+        parser.error('Either -t or -c must be defined!')
```

Comment:
Maybe we should move this check first, that is just after the `parse_args()` call.